server/service/integration_mdm_lifecycle_test.go

```diff
@@ -518,14 +518,14 @@ func (s *integrationMDMTestSuite) recordAppleHostStatus(
 	require.NoError(t, err)
 	for cmd != nil {
 		var fullCmd micromdm.CommandPayload
-		require.NoError(t, plist.Unmarshal(cmd.Raw, &fullCmd))
```

Comment:
This unmarshal in `micromdm.CommandPayload` fails if it encounters unknown request type, and now that the cron trigger runs the reconcile Apple declarations too, it can receive `DeclarativeManagement` commands. This "fixes" it in that it will only unmarshal if it is the expected request type.